run.py

```diff
@@ -454,12 +454,15 @@ def calculate_average_read_length(file_path):
             rrna_reads_dict[input_filename] / subset_reads_dict[input_filename]
             for input_filename in total_reads_dict
         ]
-
         # Use the total number of reads for each input as weights
         weights = list(total_reads_dict.values())
-        weighted_rrna_fraction = np.average(
-            fractions_rrna_in_subset, weights=weights
-        )
+        try:
+            weighted_rrna_fraction = np.average(
+                fractions_rrna_in_subset, weights=weights
+            )
+        except:
```

Comment:
Don't use bare `except`: instead list the specific exception you want to catch:

```
except FooException:
   ...
```

Reply:
fixed